src/prefect/orion/database/migrations/versions/postgresql/2023_01_25_132056_9b53afa4c332_rename_data_column.py

```diff
@@ -0,0 +1,93 @@
+"""Create temporary _data column
+
+Revision ID: 9b53afa4c332
+Revises: 9b53afa4c331
+Create Date: 2023-01-25 13:20:56.808015
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "9b53afa4c332"
+down_revision = "9b53afa4c331"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    with op.batch_alter_table("flow_run_state", schema=None) as batch_op:
+        batch_op.add_column(sa.Column("has_data", sa.Boolean))
+        batch_op.create_index(
+            batch_op.f("ix_flow_run_state__has_data"),
+            ["has_data"],
+            unique=False,
+        )
+
+    with op.batch_alter_table("task_run_state", schema=None) as batch_op:
+        batch_op.add_column(sa.Column("has_data", sa.Boolean))
+        batch_op.create_index(
+            batch_op.f("ix_task_run_state__has_data"),
+            ["has_data"],
+            unique=False,
+        )
+
+    def populate_flow_has_data_in_batches(batch_size):
+        return f"""
+            UPDATE flow_run_state
+            SET has_data = (data IS NOT NULL or data != 'null')
+            WHERE flow_run_state.id in (SELECT id FROM flow_run_state WHERE (has_data IS NULL) LIMIT {batch_size});
+        """
+
+    def populate_task_has_data_in_batches(batch_size):
+        return f"""
+            UPDATE task_run_state
+            SET has_data = (data IS NOT NULL or data != 'null')
+            WHERE task_run_state.id in (SELECT id FROM task_run_state WHERE (has_data IS NULL) LIMIT {batch_size});
+        """
+
+    def move_flow_data_in_batches(batch_size):
+        return f"""
+            UPDATE flow_run_state
+            SET _data = data
+            WHERE flow_run_state.id in (SELECT id FROM flow_run_state WHERE (has_data IS TRUE) AND (_data IS NULL) LIMIT {batch_size});
+        """
+
+    def move_task_data_in_batches(batch_size):
+        return f"""
+            UPDATE task_run_state
+            SET _data = data
+            WHERE task_run_state.id in (SELECT id FROM task_run_state WHERE (has_data IS TRUE) AND (_data IS NULL) LIMIT {batch_size});
```

Comment:
> AND (_data IS NULL)

Filtering on `_data` doesn't have an index to use here. Maybe we can iterate over `id` instead?